twitter_tunes/tests/test_bot.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+try:
+    import unittest.mock as mock
+except:
+    import mock
+from twitter_tunes.scripts import twitter_bot
+from twitter_tunes.tests import bot_test_vars
+import pytest
+
+
+def redis_side_effect(arg):
+        if arg == u'trends':
+            return bot_test_vars.REDIS_TRENDS
+        elif arg == u'last_tweets':
+            return bot_test_vars.REDIS_LAST_POSTS[:]
+
+
+@mock.patch('tweepy.API')
+def test_make_tweet_static_message(api):
+    """Test if bot makes tweet with a set message."""
+    mock_method = api().update_status
+    twitter_bot.make_tweet(u"more tests")
+    mock_method.assert_called_with(u"more tests")
+
+
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.set_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.get_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_bot.youtube_api.get_link')
+@mock.patch('twitter_tunes.scripts.twitter_api')
+@mock.patch('tweepy.API')
+def test_main_good(api, twitter_api, get_link, get_redis_data,
+                   set_redis_data):
+    """Test if bot makes tweet with api data.
+
+    This is what the main function would do.
+    """
+    from twitter_tunes.tests import test_twitter_api
+    test_url = u"https://www.youtube.com/watch?v=oyEuk8j8imI"
+    mock_update_status = api().update_status
+    mock_trends = twitter_api()
+    mock_trends.call_twitter_api.return_value = test_twitter_api.FINAL_OUTPUT
+    get_link.return_value = (test_url, True)
+    get_redis_data.side_effect = redis_side_effect
+
+    redis_trends = get_redis_data(u'trends')
+    trends = redis_trends[u'trends']
+    da_trend, url = twitter_bot.choose_trend(trends)
+    message = twitter_bot.create_message(da_trend, url)
+    twitter_bot.make_tweet(message)
+    mock_update_status.assert_called_with(message)
+
+
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.set_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_bot.redis_data.get_redis_data')
+@mock.patch('twitter_tunes.scripts.twitter_bot.twitter_api.call_twitter_api')
+@mock.patch('twitter_tunes.scripts.twitter_bot.youtube_api.get_link')
+@mock.patch('tweepy.API')
+def test_main_bad_twitter(api, youtube_api, call_twitter_api, get_redis_data,
+                          set_redis_data):
+    """Test if main does stuff if twitter goes horribly wrong.
+
+    Make sure it can keep going."""
```

Comment:
You aren't using `set_redis_data`, or `get_link` here.  or the tweepy API.  why mock them?  This tests some generalized bad condition, but it might be more instructive to write a series of tests that test different bad situations isolated from each other.  Does a failure in any one sub-system cause the bot to fail in the expected way?